services/tests/test_flake_detector.py

```diff
@@ -0,0 +1,207 @@
+from uuid import uuid4
+
+from test_results_parser import Outcome
+
+from database.models.core import Repository
+from database.models.reports import Test, TestInstance
+from database.tests.factories import CommitFactory, ReportFactory, UploadFactory
+from services.flake_detection import FlakeDetector, FlakeType
+from services.test_results import generate_test_id
+
+
+def create_repo(dbsession):
+    r = Repository()
+    dbsession.add(r)
+    dbsession.flush()
+    r.branch = "main"
+    dbsession.flush()
+    return r.repoid
+
+
+def create_commit(dbsession, repoid, branch):
+    c = CommitFactory()
+    dbsession.add(c)
+    dbsession.flush()
+    c.repoid = repoid
+    c.branch = branch
+    dbsession.flush()
+    return c.id_
+
+
+def create_report(dbsession, commit_id):
+    cr = ReportFactory()
+    dbsession.add(cr)
+    dbsession.flush()
+    cr.commit_id = commit_id
+    dbsession.flush()
+    return cr.id_
+
+
+def create_upload(dbsession, report_id):
+    u = UploadFactory()
+    dbsession.add(u)
+    dbsession.flush()
+
+    u.report_id = report_id
+    dbsession.flush()
+
+    return u.id_
+
+
+def create_test(dbsession, repoid, flags=None):
+    if flags is None:
+        name = str(uuid4())
+        testsuite = str(uuid4())
+        id_ = generate_test_id(repoid, testsuite, name, "")
+        t = Test(
+            id_=id_,
+            repoid=repoid,
+            testsuite=testsuite,
+            name=name,
+            flags_hash="",
+        )
+        dbsession.add(t)
+        dbsession.flush()
+        return id_
+    else:
+        name = str(uuid4())
```

Comment:
It's weird to have uncovered lines in your test files. Please either add tests that use these lines or refactor this portion to remove the unused lines